pilot/test/mock/config.go

```diff
@@ -516,12 +519,21 @@ func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, nam
 	CheckMapInvariant(store, t, namespace, n)
 
 	log.Infof("Waiting till all events are received")
-	util.Eventually(func() bool {
-		lock.Lock()
-		defer lock.Unlock()
-		return added == n && deleted == n
-
-	}, t)
+	lock.Lock()
```

Comment:
Should the lock/unlock be in the default below? Otherwise, you're holding the lock the entire time.

Reply:
good catch! fixed it. PTAL